modules/mono/glue/GodotSharp/GodotSharp/Core/Variant.cs

```diff
@@ -7,6 +7,59 @@ namespace Godot;
 
 #nullable enable
 
+/// <summary>
+/// <para>The most important data type in Godot.</para>
+/// <para>In computer programming, a Variant class is a class that is designed to store a variety of other types. Dynamic programming languages like PHP, Lua, JavaScript and GDScript like to use them to store variables' data on the backend. With these Variants, properties are able to change value types freely.</para>
+/// <para><code>
+/// // C# is statically typed. Once a variable has a type it cannot be changed. You can use the `var` keyword to let the compiler infer the type automatically.
+/// var foo = 2; // Foo is a 32-bit integer (int). Be cautious, integers in GDScript are 64-bit and the direct C# equivalent is `long`.
+/// // foo = "foo was and will always be an integer. It cannot be turned into a string!";
+/// var boo = "Boo is a string!";
+/// var ref = new RefCounted(); // var is especially useful when used together with a constructor.
+///
+/// // Godot also provides a Variant type that works like a union of all the Variant-compatible types.
+/// Variant fooVar = 2; // fooVar is dynamically an integer (stored as a `long` in the Variant type).
+/// fooVar = "Now fooVar is a string!";
+/// fooVar = new RefCounted(); // fooVar is a GodotObject.
+/// </code></para>
+/// <para>Godot tracks all scripting API variables within Variants. Without even realizing it, you use Variants all the time. When a particular language enforces its own rules for keeping data typed, then that language is applying its own custom logic over the base Variant scripting API.</para>
+/// <para>C# is statically typed, but uses its own implementation of the <see cref="Variant"/> type in place of Godot's Variant class when it needs to represent a dynamic value. A <see cref="Variant"/> can be assigned any compatible type implicitly but converting requires an explicit cast.</para>
+/// The <see cref="VariantType"/> function returns the enumerated value of the Variant type stored in the current variable (see <see cref="Type"/>).
+/// <para><code>
+/// Variant foo = 2;
+/// switch (foo.VariantType)
+/// {
+///     case Variant.Type.Nil:
+///         GD.Print("foo is null");
+///         break;
+///     case Variant.Type.Int:
+///         GD.Print("foo is an integer");
+///         break;
+///     case Variant.Type.Object:
+///         // Note that Objects are their own special category.
+///         // You can convert a Variant to a GodotObject and use reflection to get its name.
+///         GD.Print($"foo is a(n) {foo.AsGodotObject().GetType().Name}");
+///         break;
+/// }
+/// </code></para>
+/// <para>A Variant takes up only 20 bytes and can store almost any engine datatype inside of it. Variants are rarely used to hold information for long periods of time. Instead, they are used mainly for communication, editing, serialization and moving data around.</para>
```

Comment:
Variant takes 24 bytes (40 bytes in double precision builds) these days, the Variant documentation is a bit oudated.  The size should probably be updated in both places. But I'm not sure if putting a pretty niche implementation detail like that in the documentation is a good idea in the first place.

Reply:
Maybe, but not gonna fuss over it in this PR; I'll just update the 20 byte value to 24 for now

Reply:
Thought on it some more, and added a note about the double-precision change as well. I'll also make a seperate PR for updating the documentation itself 👍